text/0000-float-semantics.md

```diff
@@ -0,0 +1,255 @@
+- Feature Name: `float_semantics`
```

Comment:
Maybe it should be `float_nan_semantics` instead? There is more to float semantics than just NaNs, like for example there could be RFCs about changing Rust from IEEE 754-2008 to IEE 754-2019 (see the maximum vs maxnum thread in https://github.com/rust-lang/rust/issues/83984 for some discussion on this for example).

Otherwise this reminds me of #1857 which if you read the RFC only stabilized a little part of Rust's drop order logic, but got me confused at the start, making me think that say `&&` chain drop order was stable (it was not and I changed it).

Reply:
We also say that outside NaNs we exactly match IEEE. In particular, we will use exactly the precision declared in the source. In particular, this RFC explicitly says we do *not* do what https://github.com/rust-lang/rfcs/pull/2686 proposed.

So I think the name is adequate.